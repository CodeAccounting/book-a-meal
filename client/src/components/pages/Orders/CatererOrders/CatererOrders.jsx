import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import OrderPill from '../../../shared/OrderPill';
import View from '../../../shared/View';
import { userPropTypes, catererOrderObjPropTypes } from '../../../../helpers/proptypes';
import './Orders.scss';

/**
 * @exports
 * @class CatererOrders
 * @extends Component
 * @classdesc Creates CatererOrders Component
 * @returns {JSX} CatererOrders Component
 */
class CatererOrders extends Component {
  static propTypes = {
    ...userPropTypes,
    orders: PropTypes.arrayOf(catererOrderObjPropTypes).isRequired,
    isFetching: PropTypes.bool.isRequired,
    logout: PropTypes.func.isRequired,
    fetchOrders: PropTypes.func.isRequired,
  }

  /**
   * @memberof CatererOrders
   * @returns {JSX} CatererOrders Component
  */
  componentDidMount() {
    this.fetchOrders();
  }

  fetchOrders = () => {
    this.props.fetchOrders();
  }

  /**
   * @memberof CatererOrders
   * @returns {JSX} CatererOrders Component
   */
  renderCatererOrders = () => (
    <div className="main-wrapper dashboard">
      <div className="order-history">
        <div className="page-heading">
          <h2>Order History</h2>
          <hr />
        </div>
        {this.props.orders.length === 0 && <p className="text-center">You Have No Orders</p>}
        {this.props.orders.length !== 0 &&
          <div className="pills">
            {this.props.orders.map(order =>
              <OrderPill key={order.id} order={order} />)}
          </div>}
      </div>
    </div>
  )


  /**
   * @memberof CatererOrders
* @returns {JSX} CatererOrders Component
      */
  render() {
    const { user, logout, isFetching } = this.props;

    return (
      <Fragment>
        <View user={user} logout={logout} type="orders" showTime isFetching={isFetching}>
          {this.renderCatererOrders()}
        </View>
      </Fragment>
    );
  }
}

export default CatererOrders;
