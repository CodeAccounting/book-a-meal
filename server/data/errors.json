{
  "401": "Unauthorized",
  "403": "Forbidden",
  "404": "Not Found",
  "422": "Validation Error",
  "500": "Internal Server Error",
  "empty": "Empty PUT Requests Are Not Allowed"
}