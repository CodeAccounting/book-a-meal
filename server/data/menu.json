[
  {
    "userId": "8356954a-9a42-4616-8079-887a73455a7f",
     "menuId": "a9fa6cb3-9f5e-46fa-b641-388f898ca824",
     "date": "2018-05-06",
     "meals": [
        "81211c24-51c0-46ec-b1e0-18db55880958",
        "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
        "baa0412a-d167-4d2b-b1d8-404cb8f02631",
        "f9eb7652-125a-4bcb-ad81-02f84901cdc3"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  },
  {
    "userId": "8356954a-9a42-4616-8079-887a73455a7f",
     "menuId": "1adfcfe7-c66d-42d2-82fd-39c1decd290a",
     "date": "2018-04-06",
     "meals": [
        "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
        "3ec802c6-0a32-4d29-b27b-42e0f4b532dd"
     ],
     "created": "2018-04-06T14:34:90.000",
     "updated": "2018-04-06T14:34:90.000"
  },
  {
    "userId": "ac1b253c-6b33-439b-ab6f-805a4fdd2e05",
     "menuId": "3a4fb034-6e52-4bde-9abc-91119a39478d",
     "date": "2018-05-07",
     "meals": [
        "72a3417e-45c8-4559-8b74-8b5a61be8614"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  },
  {
    "userId": "ac1b253c-6b33-439b-ab6f-805a4fdd2e05",
     "menuId": "f43f3d49-a6c9-476d-b65a-1af772ff0f36",
     "date": "2018-04-06",
     "meals": [
        "72a3417e-45c8-4559-8b74-8b5a61be8614",
        "46ced7aa-eed5-4462-b2c0-153f31589bdd"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  },
  {
     "userId": "8356954a-9a42-4616-8079-887a73455a7f",
     "menuId": "15421f7a-0f82-4802-b215-e0e8efb6bfb3",
     "date": "2018-04-27",
     "meals": [
        "91b6e41c-0972-4ac5-86da-4ac1f5226e83",
        "f9eb7652-125a-4bcb-ad81-02f84901cdc3",
        "8a65538d-f862-420e-bcdc-80743df06578"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  }
]