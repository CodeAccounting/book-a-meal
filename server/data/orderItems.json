[
  {
     "id": 1,
     "mealId": "46ced7aa-eed5-4462-b2c0-153f31589bdd",
     "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
     "quantity": 2,
     "createdAt": "2018-04-06T14:34:90.000",
     "updatedAt": "2018-04-06T14:34:90.000"
  },
  {
     "id": 2,
     "mealId": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
     "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
     "quantity": 1,
     "createdAt": "2018-04-06T14:34:90.000",
     "updatedAt": "2018-04-06T14:34:90.000"
  }, 
  {
    "id": 3,
    "mealId": "72a3417e-45c8-4559-8b74-8b5a61be8614",
    "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
    "quantity": 2,
    "createdAt": "2018-04-06T14:34:90.000",
    "updatedAt": "2018-04-06T14:34:90.000"
  },
  {
    "id": 4,
    "mealId": "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
    "orderId": "6ed0963f-9663-4fe2-8ad4-2f06c6294482",
    "quantity": 2,
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000"
  },
  {
    "id": 5,
    "mealId": "baa0412a-d167-4d2b-b1d8-404cb8f02631",
    "orderId": "6ed0963f-9663-4fe2-8ad4-2f06c6294482",
    "quantity": 2,
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000"
  }
]