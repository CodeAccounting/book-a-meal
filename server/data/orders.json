[
  {
     "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
     "created": "2018-04-06T14:34:90.000",
     "updated": "2018-04-06T14:34:90.000",
     "mealId": "46ced7aa-eed5-4462-b2c0-153f31589bdd",
     "menuId": "f43f3d49-a6c9-476d-b65a-1af772ff0f36",
     "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  },
  {
     "orderId": "ce228787-f939-40a0-bfd3-6607ca8d2e53",
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000",
     "mealId": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
     "menuId": "a9fa6cb3-9f5e-46fa-b641-388f898ca824",
     "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  },
  {
     "orderId": "e544248c-145c-4145-b165-239658857637",
     "created": "2018-05-07T14:34:90.000",
     "updated": "2018-05-07T14:34:90.000",
     "mealId": "72a3417e-45c8-4559-8b74-8b5a61be8614",
     "menuId": "3a4fb034-6e52-4bde-9abc-91119a39478d",
     "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  },
  {
     "orderId": "e5508b87-7975-493d-a900-3d47a69dad03",
     "created": "2018-04-06T14:34:90.000",
     "updated": "2018-04-06T14:34:90.000",
     "mealId": "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
     "menuId": "1adfcfe7-c66d-42d2-82fd-39c1decd290a",
     "userId": "e42c7ce0-958e-4368-adcb-e5d97dbbe5b7",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  },
  {
     "orderId": "6ed0963f-9663-4fe2-8ad4-2f06c6294482",
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000",
     "mealId": "baa0412a-d167-4d2b-b1d8-404cb8f02631",
     "menuId": "a9fa6cb3-9f5e-46fa-b641-388f898ca824",
     "userId": "e42c7ce0-958e-4368-adcb-e5d97dbbe5b7",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  }
]