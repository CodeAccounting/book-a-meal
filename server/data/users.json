[
  {
    "userId": "8356954a-9a42-4616-8079-887a73455a7f",
    "username": null,
    "businessName": "FoodCircle",
    "email": "food@circle.com",
    "password": "foodcircle",
    "passwordHash": "23456789dtfgyuhio567890",
    "businessPhone": "+2348134567890",
    "businessAddress": "4, Church Street, Yaba",
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000",
    "role": "caterer"
  },
  {
    "userId": "61bb8f8d-3b59-4294-acbc-16623818c391",
    "username": "Iveren",
    "businessName": null,
    "email": "iveren@shaguy.com",
    "password": "iverenshaguy",
    "passwordHash": "23456789dtfgyuhio567890",
    "businessPhone": null,
    "businessAddress": null,
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000",
    "role": "user"
  },
  {
     "userId": "e42c7ce0-958e-4368-adcb-e5d97dbbe5b7",
     "username": "Olisa",
     "businessName": null,
     "email": "olisa@emodi.com",
     "password": "olisaemodi",
     "passwordHash": "23456789dtfgyuhio567890",
     "businessPhone": null,
     "businessAddress": null,
     "createdAt": "2018-05-06T14:34:90.000",
     "updatedAt": "2018-05-06T14:34:90.000",
     "role": "user"
  },
  {
     "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
     "username": "Emiola",
     "businessName": null,
     "email": "emiola@olasanmi.com",
     "password": "emiolaolasanmi",
     "passwordHash": "23456789dtfgyuhio567890",
     "businessPhone": null,
     "businessAddress": null,
     "createdAt": "2018-05-06T14:34:90.000",
     "updatedAt": "2018-05-06T14:34:90.000",
     "role": "user"
  },
  {
    "userId": "ac1b253c-6b33-439b-ab6f-805a4fdd2e05",
    "username": null,
    "businessName": "BellyFill",
    "email": "belly@fill.com",
    "password": "bellyfil",
    "passwordHash": "23456789dtfgyuhio567890",
    "businessPhone": "+2348134567890",
    "businessAddress": "4, Church Street, Yaba",
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000",
    "role": "caterer"
  }
]