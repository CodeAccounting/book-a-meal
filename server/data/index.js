import users from './users.json';
import meals from './meals.json';
import menu from './menu.json';
import orders from './orders.json';
import errors from './errors.json';

export default {
  users, meals, menu, orders, errors
};
