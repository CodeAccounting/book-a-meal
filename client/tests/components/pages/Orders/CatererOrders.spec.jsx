import React from 'react';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import CatererOrders from '../../../../src/components/pages/Orders/CatererOrders/CatererOrders';
import ConnectedCatererOrders from '../../../../src/components/pages/Orders/CatererOrders';
import { caterer, caterersOrdersObj, initialValues } from '../../../setup/data';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({ ...initialValues });

const { now } = Date;

describe('Orders', () => {
  beforeAll(() => {
    Date.now = jest.fn(() => 0);
  });

  afterAll(() => {
    Date.now = now;
  });

  it('renders correctly when not fetching', () => {
    const shallowWrapper = shallow(<CatererOrders
      user={caterer}
      logout={jest.fn()}
      fetchOrders={jest.fn()}
      orders={caterersOrdersObj.orders}
      isFetching={false}
    />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('OrderPill')).toBeTruthy();
  });
  it('renders Preloader when fetching', () => {
    const shallowWrapper = shallow(<CatererOrders
      user={caterer}
      logout={jest.fn()}
      fetchOrders={jest.fn()}
      orders={caterersOrdersObj.orders}
      isFetching
    />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('Preloader')).toBeTruthy();
  });

  it('renders message when not fetching and there are no orders', () => {
    const shallowWrapper = shallow(<CatererOrders
      user={caterer}
      logout={jest.fn()}
      fetchOrders={jest.fn()}
      orders={[]}
      isFetching
    />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('p').text()).toEqual('You Have No Orders');
  });

  it('renders connected component', () => {
    const comp = (
      <Provider store={store}>
        <ConnectedCatererOrders
          user={caterer}
          isFetching={false}
        />
      </Provider>);

    const wrapper = mount(comp, rrcMock.get());

    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
