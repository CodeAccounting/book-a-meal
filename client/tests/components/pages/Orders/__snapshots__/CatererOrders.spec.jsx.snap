// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders renders Preloader when fetching 1`] = `
<React.Fragment>
  <CatererView
    isFetching={true}
    logout={[MockFunction]}
    showTime={true}
    type="orders"
    user={
      Object {
        "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
        "businessName": "Food Circle",
        "businessPhoneNo": "+2348166557788",
        "email": "food@circle.com",
        "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
        "role": "caterer",
      }
    }
  >
    <div
      className="main-wrapper dashboard"
    >
      <div
        className="order-history"
      >
        <div
          className="page-heading"
        >
          <h2>
            Order History
          </h2>
          <hr />
        </div>
        <div
          className="pills"
        >
          <OrderPill
            key="fb097bde-5959-45ff-8e21-51184fa60c25"
            order={
              Object {
                "createdAt": "2018-04-06T00:47:03.687Z",
                "customer": Object {
                  "email": "emiola@olasanmi.com",
                  "firstname": "Emiola",
                  "lastname": "Olasanmi",
                },
                "deliveryAddress": "4, Church Street, Yaba",
                "deliveryPhoneNo": "+2348134567890",
                "id": "fb097bde-5959-45ff-8e21-51184fa60c25",
                "meals": Array [
                  Object {
                    "delivered": true,
                    "description": "Sharwama contains no animal products, perfect for dieters",
                    "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                    "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                    "price": "1200.00",
                    "quantity": 1,
                    "title": "Vegetable Sharwama and Guava Smoothie",
                    "vegetarian": true,
                  },
                ],
                "status": "delivered",
                "updatedAt": "2018-04-06T00:47:03.687Z",
              }
            }
          />
          <OrderPill
            key="ce228787-f939-40a0-bfd3-6607ca8d2e53"
            order={
              Object {
                "createdAt": "2018-05-01T00:47:03.687Z",
                "customer": Object {
                  "email": "olisa@emodi.com",
                  "firstname": "Olisa",
                  "lastname": "Emodi",
                },
                "deliveryAddress": "4, Church Street, Yaba",
                "deliveryPhoneNo": "+2348134567890",
                "id": "ce228787-f939-40a0-bfd3-6607ca8d2e53",
                "meals": Array [
                  Object {
                    "delivered": true,
                    "description": "Meal contains 2 pieces of beef and other assorted meat and fish products",
                    "id": "baa0412a-d167-4d2b-b1d8-404cb8f02631",
                    "imageURL": "https://static.pulse.ng/img/incoming/origs7167742/5270485143-w980-h640/Pounded-yam-and-Egusi-soup.jpg",
                    "price": "2000.00",
                    "quantity": 2,
                    "title": "Semo/Wheat and Egusi Soup",
                    "vegetarian": false,
                  },
                  Object {
                    "delivered": true,
                    "description": "Meal contains 2 pieces of beef and other assorted meat and fish products",
                    "id": "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
                    "imageURL": "http://www.kalakutahgrills.com/wp-content/uploads/2017/02/my-pics-group-3-007.jpg",
                    "price": "2000.00",
                    "quantity": 2,
                    "title": "Eba and Ogbono Soup",
                    "vegetarian": false,
                  },
                ],
                "status": "delivered",
                "updatedAt": "2018-05-01T00:47:03.687Z",
              }
            }
          />
          <OrderPill
            key="fb097bde-5959-45ff-8e21-51184fa60c26"
            order={
              Object {
                "createdAt": "2018-05-29T00:47:03.687Z",
                "customer": Object {
                  "email": "emiola@olasanmi.com",
                  "firstname": "Emiola",
                  "lastname": "Olasanmi",
                },
                "deliveryAddress": "4, Church Street, Yaba",
                "deliveryPhoneNo": "+2348134567890",
                "id": "fb097bde-5959-45ff-8e21-51184fa60c26",
                "meals": Array [
                  Object {
                    "delivered": false,
                    "description": "Sharwama contains no animal products, perfect for dieters",
                    "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                    "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                    "price": "1200.00",
                    "quantity": 1,
                    "title": "Vegetable Sharwama and Guava Smoothie",
                    "vegetarian": true,
                  },
                ],
                "status": "pending",
                "updatedAt": "2018-05-29T00:47:03.687Z",
              }
            }
          />
          <OrderPill
            key="fb097bde-5959-45ff-8e21-51184fa60c35"
            order={
              Object {
                "createdAt": "2018-05-27T00:47:03.687Z",
                "customer": Object {
                  "email": "emiola@olasanmi.com",
                  "firstname": "Emiola",
                  "lastname": "Olasanmi",
                },
                "deliveryAddress": "4, Church Street, Yaba",
                "deliveryPhoneNo": "+2348134567890",
                "id": "fb097bde-5959-45ff-8e21-51184fa60c35",
                "meals": Array [
                  Object {
                    "delivered": false,
                    "description": "Sharwama contains no animal products, perfect for dieters",
                    "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                    "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                    "price": "1200.00",
                    "quantity": 1,
                    "title": "Vegetable Sharwama and Guava Smoothie",
                    "vegetarian": true,
                  },
                ],
                "status": "canceled",
                "updatedAt": "2018-05-27T00:47:03.687Z",
              }
            }
          />
        </div>
      </div>
    </div>
  </CatererView>
</React.Fragment>
`;

exports[`Orders renders connected component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(CatererOrders)
    isFetching={false}
    user={
      Object {
        "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
        "businessName": "Food Circle",
        "businessPhoneNo": "+2348166557788",
        "email": "food@circle.com",
        "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
        "role": "caterer",
      }
    }
  >
    <CatererOrders
      deliverOrder={[Function]}
      fetchOrders={[Function]}
      isFetching={false}
      logout={[Function]}
      orders={Array []}
      user={
        Object {
          "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
          "businessName": "Food Circle",
          "businessPhoneNo": "+2348166557788",
          "email": "food@circle.com",
          "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
          "role": "caterer",
        }
      }
    >
      <CatererView
        isFetching={false}
        logout={[Function]}
        showTime={true}
        type="orders"
        user={
          Object {
            "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
            "businessName": "Food Circle",
            "businessPhoneNo": "+2348166557788",
            "email": "food@circle.com",
            "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
            "role": "caterer",
          }
        }
      >
        <div
          className="admin"
        >
          <Header
            dateType="orders"
            showTime={true}
            type="caterer"
          >
            <div
              className="header"
            >
              <div
                className="navbar"
              >
                <Connect(CatererHeader)
                  dateType="orders"
                  showTime={true}
                >
                  <CatererHeader
                    currentDay="1970-01-01"
                    dateType="orders"
                    dispatch={[Function]}
                    showTime={true}
                  >
                    <div
                      className="menu-control d-none-md"
                    >
                      <a
                        className="d-none-md"
                        href="#sidenav"
                      >
                        <img
                          alt="toggler"
                          src="./img/toggler.png"
                        />
                      </a>
                    </div>
                    <div
                      className="page-title d-flex-md"
                    >
                      <div
                        className="date"
                      >
                        <h2>
                          Thursday, 1st January 1970
                              
                          00:00
                        </h2>
                      </div>
                    </div>
                  </CatererHeader>
                </Connect(CatererHeader)>
              </div>
            </div>
          </Header>
          <div
            className="content"
          >
            <SideNav
              active="orders"
              logout={[Function]}
              user={
                Object {
                  "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
                  "businessName": "Food Circle",
                  "businessPhoneNo": "+2348166557788",
                  "email": "food@circle.com",
                  "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                  "role": "caterer",
                }
              }
            >
              <div
                className="sidenav"
                id="sidenav"
              >
                <div
                  className="sidenav-content"
                >
                  <div
                    className="sidenav-header"
                  >
                    <div
                      className="close d-none-md"
                    >
                      <a
                        href="#"
                      >
                        ×
                      </a>
                       
                    </div>
                    <div
                      className="sidenav-title"
                    >
                      <h3>
                        <a
                          href="/"
                        >
                          BOOK-A-MEAL
                        </a>
                      </h3>
                      <div
                        className="username-circle"
                      >
                        <p>
                          F
                        </p>
                      </div>
                      <p>
                        Food Circle
                      </p>
                      <span>
                        <Link
                          className={null}
                          clickHandler={[Function]}
                          id={null}
                        >
                          <button
                            className="link-btn null"
                            id={null}
                            onClick={[Function]}
                          >
                            Logout
                          </button>
                        </Link>
                      </span>
                    </div>
                  </div>
                  <div
                    className="sidenav-body"
                  >
                    <a
                      className="menu-item false"
                      href="/"
                    >
                      Dashboard
                    </a>
                    <a
                      className="menu-item false"
                      href="/meals"
                    >
                      Meals
                    </a>
                    <a
                      className="menu-item false"
                      href="/menu"
                    >
                      Menu
                    </a>
                    <a
                      className="menu-item active"
                      href="/orders"
                    >
                      Order History
                    </a>
                  </div>
                </div>
              </div>
            </SideNav>
            <div
              className="content-wrapper"
              id="has-modal"
            >
              <div
                className="main-wrapper dashboard"
              >
                <div
                  className="order-history"
                >
                  <div
                    className="page-heading"
                  >
                    <h2>
                      Order History
                    </h2>
                    <hr />
                  </div>
                  <p
                    className="text-center"
                  >
                    You Have No Orders
                  </p>
                </div>
              </div>
            </div>
          </div>
          <Footer>
            <div
              className="footer"
            >
              <p>
                © 2018. Iveren Shaguy. All images were gotten from Google and are not mine.
              </p>
            </div>
          </Footer>
        </div>
      </CatererView>
    </CatererOrders>
  </Connect(CatererOrders)>
</Provider>
`;

exports[`Orders renders correctly when not fetching 1`] = `
<React.Fragment>
  <CatererView
    isFetching={false}
    logout={[MockFunction]}
    showTime={true}
    type="orders"
    user={
      Object {
        "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
        "businessName": "Food Circle",
        "businessPhoneNo": "+2348166557788",
        "email": "food@circle.com",
        "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
        "role": "caterer",
      }
    }
  >
    <div
      className="main-wrapper dashboard"
    >
      <div
        className="order-history"
      >
        <div
          className="page-heading"
        >
          <h2>
            Order History
          </h2>
          <hr />
        </div>
        <div
          className="pills"
        >
          <OrderPill
            key="fb097bde-5959-45ff-8e21-51184fa60c25"
            order={
              Object {
                "createdAt": "2018-04-06T00:47:03.687Z",
                "customer": Object {
                  "email": "emiola@olasanmi.com",
                  "firstname": "Emiola",
                  "lastname": "Olasanmi",
                },
                "deliveryAddress": "4, Church Street, Yaba",
                "deliveryPhoneNo": "+2348134567890",
                "id": "fb097bde-5959-45ff-8e21-51184fa60c25",
                "meals": Array [
                  Object {
                    "delivered": true,
                    "description": "Sharwama contains no animal products, perfect for dieters",
                    "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                    "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                    "price": "1200.00",
                    "quantity": 1,
                    "title": "Vegetable Sharwama and Guava Smoothie",
                    "vegetarian": true,
                  },
                ],
                "status": "delivered",
                "updatedAt": "2018-04-06T00:47:03.687Z",
              }
            }
          />
          <OrderPill
            key="ce228787-f939-40a0-bfd3-6607ca8d2e53"
            order={
              Object {
                "createdAt": "2018-05-01T00:47:03.687Z",
                "customer": Object {
                  "email": "olisa@emodi.com",
                  "firstname": "Olisa",
                  "lastname": "Emodi",
                },
                "deliveryAddress": "4, Church Street, Yaba",
                "deliveryPhoneNo": "+2348134567890",
                "id": "ce228787-f939-40a0-bfd3-6607ca8d2e53",
                "meals": Array [
                  Object {
                    "delivered": true,
                    "description": "Meal contains 2 pieces of beef and other assorted meat and fish products",
                    "id": "baa0412a-d167-4d2b-b1d8-404cb8f02631",
                    "imageURL": "https://static.pulse.ng/img/incoming/origs7167742/5270485143-w980-h640/Pounded-yam-and-Egusi-soup.jpg",
                    "price": "2000.00",
                    "quantity": 2,
                    "title": "Semo/Wheat and Egusi Soup",
                    "vegetarian": false,
                  },
                  Object {
                    "delivered": true,
                    "description": "Meal contains 2 pieces of beef and other assorted meat and fish products",
                    "id": "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
                    "imageURL": "http://www.kalakutahgrills.com/wp-content/uploads/2017/02/my-pics-group-3-007.jpg",
                    "price": "2000.00",
                    "quantity": 2,
                    "title": "Eba and Ogbono Soup",
                    "vegetarian": false,
                  },
                ],
                "status": "delivered",
                "updatedAt": "2018-05-01T00:47:03.687Z",
              }
            }
          />
          <OrderPill
            key="fb097bde-5959-45ff-8e21-51184fa60c26"
            order={
              Object {
                "createdAt": "2018-05-29T00:47:03.687Z",
                "customer": Object {
                  "email": "emiola@olasanmi.com",
                  "firstname": "Emiola",
                  "lastname": "Olasanmi",
                },
                "deliveryAddress": "4, Church Street, Yaba",
                "deliveryPhoneNo": "+2348134567890",
                "id": "fb097bde-5959-45ff-8e21-51184fa60c26",
                "meals": Array [
                  Object {
                    "delivered": false,
                    "description": "Sharwama contains no animal products, perfect for dieters",
                    "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                    "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                    "price": "1200.00",
                    "quantity": 1,
                    "title": "Vegetable Sharwama and Guava Smoothie",
                    "vegetarian": true,
                  },
                ],
                "status": "pending",
                "updatedAt": "2018-05-29T00:47:03.687Z",
              }
            }
          />
          <OrderPill
            key="fb097bde-5959-45ff-8e21-51184fa60c35"
            order={
              Object {
                "createdAt": "2018-05-27T00:47:03.687Z",
                "customer": Object {
                  "email": "emiola@olasanmi.com",
                  "firstname": "Emiola",
                  "lastname": "Olasanmi",
                },
                "deliveryAddress": "4, Church Street, Yaba",
                "deliveryPhoneNo": "+2348134567890",
                "id": "fb097bde-5959-45ff-8e21-51184fa60c35",
                "meals": Array [
                  Object {
                    "delivered": false,
                    "description": "Sharwama contains no animal products, perfect for dieters",
                    "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                    "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                    "price": "1200.00",
                    "quantity": 1,
                    "title": "Vegetable Sharwama and Guava Smoothie",
                    "vegetarian": true,
                  },
                ],
                "status": "canceled",
                "updatedAt": "2018-05-27T00:47:03.687Z",
              }
            }
          />
        </div>
      </div>
    </div>
  </CatererView>
</React.Fragment>
`;

exports[`Orders renders message when not fetching and there are no orders 1`] = `
<React.Fragment>
  <CatererView
    isFetching={true}
    logout={[MockFunction]}
    showTime={true}
    type="orders"
    user={
      Object {
        "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
        "businessName": "Food Circle",
        "businessPhoneNo": "+2348166557788",
        "email": "food@circle.com",
        "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
        "role": "caterer",
      }
    }
  >
    <div
      className="main-wrapper dashboard"
    >
      <div
        className="order-history"
      >
        <div
          className="page-heading"
        >
          <h2>
            Order History
          </h2>
          <hr />
        </div>
        <p
          className="text-center"
        >
          You Have No Orders
        </p>
      </div>
    </div>
  </CatererView>
</React.Fragment>
`;
